org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AbstractRegressionTest.java

```diff
@@ -4078,7 +4078,8 @@ protected void setUp() throws Exception {
 					String version = JavacCompiler.getVersion(cmdLineHeader.toString());
 					cmdLineHeader.append(" -d ");
 					cmdLineHeader.append(JAVAC_OUTPUT_DIR_NAME.indexOf(" ") != -1 ? "\"" + JAVAC_OUTPUT_DIR_NAME + "\"" : JAVAC_OUTPUT_DIR_NAME);
-					cmdLineHeader.append(" -source 1.5 -deprecation -Xlint "); // enable recommended warnings
+					String firstSupportedVersion = CompilerOptions.getFirstSupportedJavaVersion();
+					cmdLineHeader.append(" -source " + firstSupportedVersion + " -deprecation -Xlint "); // enable recommended warnings
```

Comment:
looks funny vis-a-vis variable `version` above, but is OK (`cmdLineHeader` happens to be used **only** in `BatchCompilerTest.test231_sourcepath_vs_classpath()` when run with `run.javac` arguments)